### PR TITLE
[hw,i2c,dv] Update perf test to check for DUT performance

### PR DESCRIPTION
**Pull request by crteja**
- Add enum in `env_pkg` to indicate the speed mode of I2C
- Update timing parameter constraints to match the speed mode
- Calculate bit rate at the end of test and check if value is greater than a threshold of SCL frequency

### CODE REVIEW

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (marnovandermaas) [inline hw/ip/i2c/dv/env/i2c_seq_cfg.sv:77]: This is specified as 20 in the standard right?

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (marnovandermaas) [inline hw/ip/i2c/dv/env/i2c_seq_cfg.sv:66]: I can't find this in the standard (Table 10): https://web.archive.org/web/20210813122132/https://www.nxp.com/docs/en/user-guide/UM10204.pdf
Where did you get this number from?

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (marnovandermaas) [inline hw/ip/i2c/dv/env/i2c_seq_cfg.sv:67]: Also can't find this one in the standard

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (marnovandermaas) [inline hw/ip/i2c/dv/env/i2c_seq_cfg.sv:78]: This is specified as function of V_dd, do we know that value?

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (marnovandermaas) [inline hw/ip/i2c/dv/env/i2c_seq_cfg.sv:88]: Can't find this in the spec Table 10

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (marnovandermaas) [inline hw/ip/i2c/dv/env/i2c_seq_cfg.sv:89]: Also defined as function of V_DD. Do we know this value?

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv
Comment (marnovandermaas) [inline hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv:27]: This works, but maybe a bit of a misleading calling `scl_frequency` random.

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/i2c_seq_cfg.sv:66]: Since there is no minimum value and the maximum value for `Standard` mode is 1000ns, I have taken the maximum value of highest speed `FastPlus` mode

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/i2c_seq_cfg.sv:67]: same as `TRMINSTARTARD`, used the max value of `tF` from `FastPlus` mode

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/i2c_seq_cfg.sv:77]: yes, will correct this

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/i2c_seq_cfg.sv:88]: same as [r1149107055](https://github.com/lowRISC/opentitan/pull/17675/files#r1149107055)

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/i2c_seq_cfg.sv:78]: Not sure of the operating voltage for OT, used max value of `FastPlus` mode similar to [r1149107055](https://github.com/lowRISC/opentitan/pull/17675/files#r1149107055)

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/i2c_seq_cfg.sv:89]: Not sure of the operating voltage for OT, used max value of `FastPlus` mode similar to [r1149107055](https://github.com/lowRISC/opentitan/pull/17675/files#r1149107055)

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv:27]: This will be updated later to cover non-standard frequencies such as
- Frequencies less than 100KHz for `Standard` mode
- Frequencies less than 400KHz for `Fast` mode
- Frequencies less than 1MHz for `FastPlus` mode

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (marnovandermaas) [reply hw/ip/i2c/dv/env/i2c_seq_cfg.sv:66]: Just wanted to capture what we talked about. I think it's probably better not to use the maximum of Fast-mode Plus here. For $T_R$ I think it is reasonable to use 20 for all modes and for $T_F$ we can assume $V_{DD}$ is 5.5 V.

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/i2c_seq_cfg.sv]: Updated as per [r1149385405](https://github.com/lowRISC/opentitan/pull/17675#discussion_r1149385405)

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/i2c_seq_cfg.sv]: Updated as per [r1149385405](https://github.com/lowRISC/opentitan/pull/17675#discussion_r1149385405)

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/i2c_seq_cfg.sv]: Updated as per [r1149385405](https://github.com/lowRISC/opentitan/pull/17675#discussion_r1149385405)

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/i2c_seq_cfg.sv]: Updated as per [r1149385405](https://github.com/lowRISC/opentitan/pull/17675#discussion_r1149385405)

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv
Comment (marnovandermaas) [inline hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv:59]: Do we need to add more solve orders for this constraint? For example solve `scl_period` before anything on the left hand side?

**Review by marnovandermaas — COMMENTED** [review]
We need to await discussion of parameters before merging this PR. In the meantime one comment.

**Comment by marnovandermaas** [comment]
Would you mind updating the parameters in this PR to these proposed values (pending confirmation):
image

**Comment by crteja** [comment]
> Would you mind updating the parameters in this PR to these proposed values (pending confirmation): image

Updated in [change](https://github.com/lowRISC/opentitan/compare/b1309c416daa68efcf9aac4fd7994832638e3bea..2def5d6440b705bc9ccef61aad5099c01ba7602f)

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/i2c_seq_cfg.sv]: Sure, added two macros [c02d58d](https://github.com/lowRISC/opentitan/compare/2def5d6440b705bc9ccef61aad5099c01ba7602f..c02d58d15546ab56d1e5e6f017a64739ca9504fa). tested changes with both xcelium and vcs

Comment (marnovandermaas) [reply hw/ip/i2c/dv/env/i2c_seq_cfg.sv]: Thanks!

**Review by marnovandermaas — APPROVED** [review]
Thanks for making these changes. I'm approving this PR, but lets wait for another person to review before merging.

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv
Comment (hcallahan-lowrisc) [inline hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv:12]:
```suggestion
  rand uint                   scl_frequency; //in kHz
```

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv
Comment (hcallahan-lowrisc) [inline hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv:13]:
```suggestion
  rand uint                   scl_period; // converted to register value
```

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv
Comment (hcallahan-lowrisc) [inline hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv:16]: nit : spaces around operator
(this applies to the rest of the changes too, the convention is to have at-least one space around operators.)

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv
Comment (hcallahan-lowrisc) [inline hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv:109]: Can you add a description to this task? Helps with readability. Thanks

**Review by hcallahan-lowrisc — APPROVED** [review]
Looks good, with some minor nits about formatting and comments. Thanks @crteja !

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/i2c_seq_cfg.sv]: Updated as per comments in [d18a44a](https://github.com/lowRISC/opentitan/compare/c02d58d15546ab56d1e5e6f017a64739ca9504fa..d18a44a133ba13ed227adc37d4d209c099bf7fc9)

File: hw/ip/i2c/dv/env/i2c_seq_cfg.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/i2c_seq_cfg.sv]: Updated as per comments in [d18a44a](https://github.com/lowRISC/opentitan/compare/c02d58d15546ab56d1e5e6f017a64739ca9504fa..d18a44a133ba13ed227adc37d4d209c099bf7fc9)

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv:1]: Added test description in [d18a44a](https://github.com/lowRISC/opentitan/compare/c02d58d15546ab56d1e5e6f017a64739ca9504fa..d18a44a133ba13ed227adc37d4d209c099bf7fc9)

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv]: Updated as per comments in [d18a44a](https://github.com/lowRISC/opentitan/compare/c02d58d15546ab56d1e5e6f017a64739ca9504fa..d18a44a133ba13ed227adc37d4d209c099bf7fc9)

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv]: Updated as per comments in [d18a44a](https://github.com/lowRISC/opentitan/compare/c02d58d15546ab56d1e5e6f017a64739ca9504fa..d18a44a133ba13ed227adc37d4d209c099bf7fc9)

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv]: Updated as per comments in [d18a44a](https://github.com/lowRISC/opentitan/compare/c02d58d15546ab56d1e5e6f017a64739ca9504fa..d18a44a133ba13ed227adc37d4d209c099bf7fc9)

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv:118]: Added task description in [d18a44a](https://github.com/lowRISC/opentitan/compare/c02d58d15546ab56d1e5e6f017a64739ca9504fa..d18a44a133ba13ed227adc37d4d209c099bf7fc9)

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv
Comment (marnovandermaas) [inline hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv:135]:
```suggestion
    real end_time = $realtime;
```

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv
Comment (marnovandermaas) [inline hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv:143]:
```suggestion
    bit_rate = (num_bits * (10**6)) / (end_time - start_time); // in Kbps
```

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv
Comment (marnovandermaas) [inline hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv:145]:
```suggestion
    `uvm_info(`gfn, $sformatf("bit_rate = %d Kbps", bit_rate), UVM_LOW)
```

**Review by marnovandermaas — COMMENTED** [review]
Please try to adhere to the style guide:
https://github.com/lowRISC/style-guides/blob/master/VerilogCodingStyle.md#expressions

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv]: Updated as per comments

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv]: Updated as per comments

File: hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv
Comment (crteja) [reply hw/ip/i2c/dv/env/seq_lib/i2c_host_perf_vseq.sv]: updated as per comments

**Comment by crteja** [comment]
Rebased changes to resolve merge conflicts

**Comment by marnovandermaas** [comment]
Rerunning CI because the failure mentions UART, which I don't think is affected by this PR.

**Pull request merged**
